src/mscorlib/src/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -2403,8 +2400,10 @@ public static String GenerateProgIdForType(Type type)
                 throw new ArgumentException(Environment.GetResourceString("Argument_NeedNonGenericType"), nameof(type));
             Contract.EndContractBlock();
 
+#if !FEATURE_CORECLR
             if (!RegistrationServices.TypeRequiresRegistrationHelper(type))
                 throw new ArgumentException(Environment.GetResourceString("Argument_TypeMustBeComCreatable"), nameof(type));
+#endif // FEATURE_CORECLR            
 
             IList<CustomAttributeData> cas = CustomAttributeData.GetCustomAttributes(type);
```

Comment:
Note that you will also need to delete ProgIdAttribute in corefx, and use the forwarded copy from corelib instead to make this work well.